ambry-named-mysql/src/integration-test/java/com/github/ambry/named/MySqlNamedBlobDbIntegrationTest.java

```diff
@@ -210,6 +230,79 @@ public void testExpiredBlobs() throws Exception {
         namedBlobDb.get(account.getName(), container.getName(), blobName).get());
   }
 
+  /**
+   * Test behavior with stale blob cleanup
+   */
+  @Test
+  public void testCleanupStaleBlobs() throws Exception {
+    Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+    calendar.add(Calendar.MONTH, -2);
+    long twoMonthAgo = calendar.getTimeInMillis();
+    long twoMonthAgoABitLater = twoMonthAgo + 1;
+
+    Account account = accountService.getAllAccounts().iterator().next();
+    Container container = account.getAllContainers().iterator().next();
+
+    int staleCount = 10;
+    int needCleanupCount = 0;
+    List<NamedBlobRecord> records = new ArrayList<>();
+
+    // Create stale named blob records
+    for (int i = 0; i < staleCount; i++) {
+      String blobId = getBlobId(account, container);
+      String blobName = "stale/" + i + "/more path segments--";
+      long expirationTime = i % 2 == 0 ? Utils.Infinite_Time : twoMonthAgoABitLater;
+      NamedBlobRecord record =
```

Comment:
we are creating stale named blob for Update API for now. These should be considered stale. Assume we give 5 days of retention.
Case 1. created 10 days ago, InProgress, is Largest (Don't care about the ttl and delete)
Case 2. created 10 days ago, Ready, not Largest and bigger version is ready

And these should considered good
Case 1, created 2 days ago, InProgress, is Largest (Don't care about ttl and delete)
Case 2. created 2 days ago, Ready, not largest, and bigger version is ready
Case 3. created 10 days ago, Ready, is Largest
Case 4. created 10 days ago, Ready, not Largest, but bigger version is InProgress.


Can you at least write all those cases and make sure that are treated as expected.

Reply:
Added.